File: dynamic_closure/customizations/item_attribute/item_attribute.py
from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.model.document import Document
from erpnext.controllers.item_variant import create_variant
from six import string_types
import json

def validate(doc, method=None):
	pass


@frappe.whitelist()
def publish_tags(item_attribute,tagname):
	tag_doc = frappe.new_doc("Tag")
	tag_doc.__newname = tagname
	tag_doc.description = item_attribute + " " + tagname
	tag_doc.flags.ignore_permissions = True
	tag_doc.save()
	return True

@frappe.whitelist()
def publish_all_tags(item_attribute):
	tag_doc = frappe.new_doc("Tag")
	tag_doc.__newname = "ALL-" + item_attribute
	tag_doc.description = "ALL-" + item_attribute
	tag_doc.flags.ignore_permissions = True
	tag_doc.save()

	frappe.db.commit()
	return True

File: dynamic_closure/customizations/item_attribute/item_attribute.js
frappe.ui.form.on('Item Attribute', {
	refresh: function(frm) {
		frm.fields_dict.item_attribute_values.grid.toggle_enable("publish_as_tag",(frm.doc.applicable_to_all)? false: true);
		frm.refresh_fields('item_attribute_values');
	},
	applicable_to_all: function(frm) {
		frm.fields_dict.item_attribute_values.grid.toggle_enable("publish_as_tag",(frm.doc.applicable_to_all)? false: true);
		
		frm.refresh_fields('item_attribute_values');

		frappe.call({
			method: 'dynamic_closure.customizations.item_attribute.item_attribute.publish_all_tags',
			args: {
                "item_attribute": frm.doc.name
            },
			callback: (r) => {
				if(r.message)
				{
					frappe.msgprint("Tag published successfully")
					frm.save()
				}
			}
		});
	}
});

frappe.ui.form.on('Item Attribute Value', {
	publish_as_tag: function(frm, cdt, cdn) {
		// let me = this;
		var row = locals[cdt][cdn]
		if(row.publish_as_tag)
		{
			
			frappe.call({
				method: 'dynamic_closure.customizations.item_attribute.item_attribute.publish_tags',
				args: {
	                "item_attribute": frm.doc.name,
	                "tagname": row.attribute_value
	            },
				callback: (r) => {
					if(r.message)
					{
						frappe.msgprint("Tag published successfully")
						frm.save()
					}
				}
			});
		}
	}

});

